acceptance/tests/parser_functions/puppet_lookup_cmd.rb

```diff
@@ -6,12 +6,11 @@
 
 ### @testroot = "/etc/puppetlabs"
 @testroot = master.tmpdir("#{@module_name}")
-
 @coderoot = "#{@testroot}/code"
 @confdir = "#{@testroot}/puppet"
-
-@mastername = on(master, facter('fqdn')).stdout.chomp
-@agentname = on(agent, facter('fqdn')).stdout.chomp
+@agent = [agent].flatten.find { |host| host != master }
```

Comment:
Does this mean that the test would only run on at most one non-master agent?  For the case where we would be running tests with a configuration in which more than one non-master agent is present in the configuration, would we consider it a problem that only one of those agents would be exercised?  Also, what would happen if this test were run in a single agent configuration where that one agent is running on the master?  Would the test blow up in that case?

AFAIK, no teams are currently running with either of the configurations that I mentioned above in their standard CI pipelines but it would seem good to future-proof this for cases where this might be done at some point.